src/org/jgroups/protocols/raft/InMemoryLog.java

```diff
@@ -110,111 +114,141 @@ public ByteBuffer getSnapshot() {
     }
 
     @Override
-    public synchronized long append(long index, LogEntries entries) {
-        int space_required=entries != null? entries.size() : 0;
-        long available_space=this.entries.length - 1 - last_appended;
-        if(space_required > available_space)
-            expand((int)(space_required - available_space +1));
-
-        for(LogEntry entry: entries) {
-            int idx=(int)(index-first_appended); // we cannot be more than 2^31 (max size of int) apart!
-            this.entries[idx]=entry;
-            last_appended=Math.max(last_appended, index);
-            index++;
-            if(entry.term > current_term)
-                current_term=entry.term;
+    public long append(long index, LogEntries entries) {
+        LogEntry[] elements = entries.entries.toArray(new LogEntry[entries.size()]);
+        long term = 0;
+        for (int i = 0; i < entries.size(); i++) {
+            LogEntry entry = entries.entries.get(i);
+            if (entry.term > term) term = entry.term;
         }
-        return last_appended;
+
```

Comment:
@jabolina why are you copying the entries to an array?
Also, since `LogEntries` is sorted, the highest `term` is present in the last element.

Reply:
Yeah, this copy shouldn't exist, sorry.

I can patch this and change it to remove the read-write lock (if it is safe to do so). I am using the lock because I wasn't 100% sure about the single thread.